e2e/runner/emissions.go

```diff
@@ -0,0 +1,90 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/zeta-chain/node/cmd/zetacored/config"
+	"github.com/zeta-chain/node/e2e/txserver"
+	e2eutils "github.com/zeta-chain/node/e2e/utils"
+	emissionstypes "github.com/zeta-chain/node/x/emissions/types"
+	observertypes "github.com/zeta-chain/node/x/observer/types"
+)
+
+// FundEmissionsPool funds the emissions pool on ZetaChain with the same value as used originally on mainnet (20M ZETA)
+func (r *E2ERunner) FundEmissionsPool() error {
+	r.Logger.Print("⚙️ funding the emissions pool on ZetaChain with 20M ZETA (%s)", txserver.EmissionsPoolAddress)
+
+	return r.ZetaTxServer.FundEmissionsPool(e2eutils.OperationalPolicyName, EmissionsPoolFunding)
+}
+
+// WithdrawEmissions withdraws emissions from the emission pool on ZetaChain for all observers
+// This functions uses the UserEmissionsWithdrawName to create the withdraw tx.
+// UserEmissionsWithdraw can sign the authz transactions because the necessary permissions are granted in the genesis file
+func (r *E2ERunner) WithdrawEmissions() error {
+	observerSet, err := r.ObserverClient.ObserverSet(r.Ctx, &observertypes.QueryObserverSet{})
+	if err != nil {
+		return err
+	}
+
+	for _, observer := range observerSet.Observers {
+		r.Logger.Print("🏃 Withdrawing emissions for observer %s", observer)
+		var (
+			baseDenom            = config.BaseDenom
+			queryObserverBalance = &banktypes.QueryBalanceRequest{
+				Address: observer,
+				Denom:   baseDenom,
+			}
+		)
+
+		balanceBefore, err := r.BankClient.Balance(r.Ctx, queryObserverBalance)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get balance for observer before withdrawing emissions %s", observer)
+		}
+
+		availableAmount, err := r.EmissionsClient.ShowAvailableEmissions(
+			r.Ctx,
+			&emissionstypes.QueryShowAvailableEmissionsRequest{
+				Address: observer,
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to get available emissions for observer %s: %w", observer, err)
+		}
+
+		availableCoin, err := sdk.ParseCoinNormalized(availableAmount.Amount)
+		if err != nil {
+			return fmt.Errorf("failed to parse coin amount: %w", err)
+		}
+
+		if availableCoin.Amount.IsZero() {
+			r.Logger.Print("no emissions to withdraw for observer %s", observer)
+			continue
+		}
+
+		err = r.ZetaTxServer.WithdrawAllEmissions(availableCoin.Amount, e2eutils.UserEmissionsWithdrawName, observer)
+		if err != nil {
+			return err
+		}
+
+		balanceAfter, err := r.BankClient.Balance(r.Ctx, queryObserverBalance)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get balance for observer after withdrawing emissions %s", observer)
+		}
+
+		changeInBalance := balanceAfter.Balance.Sub(*balanceBefore.Balance).Amount
+		if !changeInBalance.Equal(availableCoin.Amount) {
+			return fmt.Errorf(
+				"invalid balance change for observer %s, expected %s, got %s",
+				observer,
+				availableCoin.Amount,
+				changeInBalance,
+			)
+		}
+	}
```

Comment:
It is difficult to check the emissions pool balance or available emissions here, as there can be some emissions in the blocks in between the tx's.

Reply:
Why not withdraw all emissions by default? No need to specify a certain amount. 

That's how the commission withdrawal works for validators and I can't imagine a situation in which someone needs to withdraw some but not all emission rewards.

Reply:
We are withdrawing all the available emissions.

Reply:
We can update the emissions params to set percentage to 0 before withdrawing.
So no emission happens after

Reply:
I considered that, but we would have to unset it after the check so that tss migration and upgrade tests are successful, 
It seemed to me a bit unnecessary to do that just for this check, considering we also have unit tests .
Happy to add it though , if needed . Its trivial effort